src/smt/theory_str_noodler/theory_str_noodler_final_check.cpp

```diff
@@ -430,6 +430,9 @@ namespace smt::noodler {
 
     bool theory_str_noodler::is_underapprox_suitable(const Formula& instance, const AutAssignment& aut_ass) const {
         int ln = 0;
+        if(!this->m_conversion_todo.empty()) {
+            return false;
+        }
```

Comment:
What does this do?

Reply:
It means that underapproximation will not turn on if we have any `to_code/from_code/to_int/from_int`. Is there any reason we want to turn it off? If yes, add a comment.

Reply:
Ok, when I thought about it, it seems to me that it should work for the conversions.